### PR TITLE
Update etcd versions and add coredns and flannel

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Update etcd versions and add coredns and flannel

###### Change Log  

update or create the relevant spec files

###### Does this affect the toolchain?  

NO

### CODE REVIEW

File: SPECS/coredns/coredns-1.6.5.spec [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Source1:        %{name}-%{version}-vendor.tar.gz
+Patch0:         makefile-buildoption-commitnb-%{version}.patch
+
+BuildRequires:  golang >= 1.12
+
+%description
+CoreDNS is a fast and flexible DNS server.
+
+%prep
+%setup -q
```
Comment (MateuszMalisz) [inline SPECS/coredns/coredns-1.6.5.spec:25]: nit: Couldn't we use autosetup here?

File: SPECS/etcd/etcd-3.4.13.spec [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+BuildRequires:  git
+
+%description
+A highly-available key value store for shared configuration and service discovery.
+
+%prep
+%setup -q
+
+%build
+# Turn off auto moduling. golang 1.13 does not automatically consider the vendor folder (it does as of 1.14).
```
Comment (MateuszMalisz) [inline SPECS/etcd/etcd-3.4.13.spec:24]: Why not simply depend on golang >=1.14 then? If not, shouldn't we guard this bit to do different things for 1.14+?

File: SPECS/etcd/etcd-3.4.13.spec [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+mv %{_builddir}/%{name}-%{version}/etcdctl/README.md %{buildroot}/%{_docdir}/%{name}-%{version}/README-etcdctl.md
+mv %{_builddir}/%{name}-%{version}/etcdctl/READMEv2.md %{buildroot}/%{_docdir}/%{name}-%{version}/READMEv2-etcdctl.md
+
+install -vdm755 %{buildroot}/lib/systemd/system-preset
+echo "disable etcd.service" > %{buildroot}/lib/systemd/system-preset/50-etcd.preset
+
+cp %{SOURCE1} %{buildroot}/lib/systemd/system
+install -vdm755 %{buildroot}/var/lib/etcd
+
+%post   -p /sbin/ldconfig
```
Comment (MateuszMalisz) [inline SPECS/etcd/etcd-3.4.13.spec:60]: Shouldn't this be `%postin` instead?

File: SPECS/etcd/etcd-3.4.13.spec [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+%clean
+rm -rf %{buildroot}/*
+
+%files
+%license LICENSE
+%{_bindir}/etcd*
+/%{_docdir}/%{name}-%{version}/*
+/lib/systemd/system/etcd.service
+/lib/systemd/system-preset/50-etcd.preset
+%dir /var/lib/etcd
```
Comment (MateuszMalisz) [inline SPECS/etcd/etcd-3.4.13.spec:73]: Nit: I'm not sure if we use that anywhere, but `/var/lib` can be represented with `%{_sharedstatedir}`.

File: SPECS/flannel/flannel.spec [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+# move sources where make expect them to be (see flannel building.md)
+export GOPATH=%{our_gopath}
+mkdir -p "%{gopath_flannel_folder}"
+mv %{_builddir}/%{name}-%{version}/* %{gopath_flannel_folder}
+# but keep license file where it was to make %license macro happy
+cp %{gopath_flannel_folder}/LICENSE %{_builddir}/%{name}-%{version}
+
+# build flannel
+export TAG=v%{version}
+%ifarch x86_64
```
Comment (MateuszMalisz) [inline SPECS/flannel/flannel.spec:37]: Should we add some directives to mention that these two are the only valid architectures(e.g. BuildArch)?

**Review by MateuszMalisz — APPROVED** [review]

File: SPECS/coredns/coredns-1.6.5.spec (hunk 1 above)
Comment (nicogbg) [reply SPECS/coredns/coredns-1.6.5.spec:25]: fixed in all 3 CoreDNS spec

File: SPECS/etcd/etcd-3.4.13.spec (hunk 2 above)
Comment (nicogbg) [reply SPECS/etcd/etcd-3.4.13.spec:24]: technically etcd requires golang 1.13 and I think we'd better stick with what etcd stated (1.13+)
fyi, I just kept the existing etcd spec as it was

File: SPECS/etcd/etcd-3.4.13.spec (hunk 4 above)
Comment (nicogbg) [reply SPECS/etcd/etcd-3.4.13.spec:73]: changed in both etcd spec

File: SPECS/etcd/etcd-3.4.13.spec (hunk 3 above)
Comment (nicogbg) [reply SPECS/etcd/etcd-3.4.13.spec:60]: keep etcd spec as it was

File: SPECS/flannel/flannel.spec (hunk 5 above)
Comment (nicogbg) [reply SPECS/flannel/flannel.spec:37]: keep as it is since CBL-Mariner is only AMD64 and ARM64

**Pull request merged**